src/Service/CaseManager.php

```diff
@@ -99,4 +103,47 @@ public function newCase(CaseEntity $caseEntity, Board $board): CaseEntity
 
         return $caseEntity;
     }
+
+    public function updateDeadlineBooleans($isDryRun)
+    {
+        // We do this in batches as there might be a large number of cases
+        // @see https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/batch-processing.html#iterating-results
+        $batchSize = 20;
+        $i = 1;
+        $query = $this->caseRepository->createQueryBuilder('c')->getQuery();
+
+        $today = new DateTime('today');
+
+        $this->logger->info('Today: '.$today->format('d/m/Y'));
+
+        foreach ($query->toIterable() as $case) {
+            assert($case instanceof CaseEntity);
+            $hearingDeadline = $case->getFinishHearingDeadline();
+            $processingDeadline = $case->getFinishProcessingDeadline();
+
+            if ($hearingDeadline < $today) {
+                $this->logger->info('Changing case '.$case->getCaseNumber().' with hearing deadline: '.$hearingDeadline->format('d/m/Y'));
+
+                if (!$isDryRun) {
+                    $case->setHasReachedHearingDeadline(true);
+                }
+
+                if ($processingDeadline < $today) {
+                    $this->logger->info('Changing case '.$case->getCaseNumber().' with processing deadline: '.$hearingDeadline->format('d/m/Y'));
+
+                    if (!$isDryRun) {
+                        $case->setHasReachedProcessingDeadline(true);
+                    }
+                }
+            }
+            ++$i;
+
+            if (($i % $batchSize) === 0 && !$isDryRun) {
+                $this->entityManager->flush(); // Executes all updates.
+                $this->entityManager->clear(); // Detaches all objects from Doctrine!
+            }
+        }
+
+        $this->entityManager->flush();
```

Comment:
Should this be executed if `$isDryRun` is `FALSE`?

Reply:
Nope, fixed.